src/types.rs

```diff
@@ -240,6 +240,60 @@ impl BlockTime {
     }
 }
 
+/// Balance differentiated in various categories
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Default)]
```

Comment:
I think here you should derive `Eq` as well, since this structure has the required properties:

> In addition to a == b and a != b being strict inverses, the equality must be (for all a, b and c):
> - reflexive: a == a;
> - symmetric: a == b implies b == a; and
> - transitive: a == b and b == c implies a == c.